ribs/archives/_dns_archive.py

```diff
@@ -380,47 +331,19 @@ def add(
         cur_size = len(self)
         if cur_size > 0:
             cur = self._store.data(return_type="dict")
-        else:
-            cur = {
-                name: np.empty((0, *desc[0]), dtype=desc[1])
-                for name, desc in self._store.field_desc.items()
-            }  # type: ignore[index]
-
-        # Combine.
-        combined = {}
-        for name in self._store.field_list:
-            combined[name] = (
-                np.concatenate((cur[name], data[name]), axis=0)
-                if name in data
-                else cur[name]
-            )
 
-        # Compute DNS over union.
-        measures = combined["measures"]
-        objs = combined["objective"]
-
-        # Pairwise distances.
-        if measures.shape[0] == 0:
-            dns_scores = np.array([], dtype=self.dtypes["measures"])  # pragma: no cover
+            # Combine.
+            combined = {}
+            for name in self._store.field_list:
+                combined[name] = (
+                    np.concatenate((cur[name], data[name]), axis=0)
+                    if name in data
```

Comment:
Maybe we should throw an error instead for the case when `if name not in data`? I don't imagine your code will behave very nicely if you continue to use `combined` with some entries being longer than others.